bip-XXXX.mediawiki

```diff
@@ -0,0 +1,84 @@
+<pre>
+  BIP: XXXX
+  Layer: Applications
+  Title: DNS Payment Instructions
+  Author: Matt Corallo <bipxxx@bluematt.me>
+          Bastien (once I confirm he wants to be listed here, but its his idea)
+  Comments-Summary: No comments yet.
+  Status: Draft
+  Type: Standards Track
+  Created: 2024-02-10
+</pre>
+
+==Abstract==
+This BIP proposes a standard format for encoding [[bip-0021.mediawiki|BIP 21]] URI schemes in DNS TXT records.
+
+==Motivation==
```

Comment:
It would be very helpful if this section was extended with a number of concrete use cases and also mention which use cases are not covered or not a good fit.

At first this seems like an interesting solution to display domain names instead of addresses on hardware wallets for improved security, but for exchanges/brokers/merchants, they'd need a unique address per user and per deposit. For this use-case this does not seem to be a good fit. Downsides / blockers could be:

- if it is generated on the fly during e.g. a deposit action on an exchange or a shop checkout of a merchant, the TXT record might not propagate quickly enough for the lookup to succeed
- publishing a new address after the previous one has been used also might not propagate in time to prevent re-using an address
- bad privacy for the merchant/exchange
- not quite sure if it is practical to have a lot of txt records for services that serve a lot of users

Reply:
> That's okay. Address reuse prevention is probabilistic anyway.

`.well-known` approach can prevent address reuse completely, if HTTP server generates new address for every request.